core/src/epicli/data/common/ansible/playbooks/roles/upgrade/tasks/kubernetes/patch-kubeadm-config.yml

```diff
@@ -0,0 +1,130 @@
+  block:
+    - name: Check the kubeadm-config.yml file
+      stat:
+        path: &kubeadm-config-yml /etc/kubeadm/kubeadm-config.yml
+        get_attributes: false
+        get_checksum: false
+        get_mime: false
+      register: stat_kubeadm_config
+
+- when:
+    - stat_etcd_encryption_config.stat.exists
+    - stat_kubeadm_config.stat.exists
+  block:
+    - name: Load contents of the kubeadm-config.yml file
+      slurp:
+        path: *kubeadm-config-yml
+      register: slurp_kubeadm_config
+
+    - name: Save modified contents of the kubeadm-config.yml file
+      copy:
+        dest: *kubeadm-config-yml
+
+        # Save all documents.
+        content: |
+          {% for document in _documents_updated %}
+          ---
+          {{ document | to_nice_yaml(indent=2) }}
+          {% endfor -%}
+
+      vars:
+        # Parse yaml payload (remove empty documents).
+        _documents: >-
+          {{ slurp_kubeadm_config.content | b64decode
+                                          | from_yaml_all
+                                          | select
+                                          | list }}
+        # Prepare the patch.
+        # In this patch we include location of the etcd encryption config file.
+        # If it is not included, then the etcd encryption feature becomes disabled/broken.
+        # If it is not present on a cluster that has kube-system secrets already encrypted, then
+        # it may cause any upgrade attempt to freeze for a very long time (in Epiphany it has been reported to be even up to 8 hours).
+        _update:
+          apiServer:
+            extraArgs:
+              encryption-provider-config: *etc-encryption-conf
+
+        # Process all documents (returns a list of dictionaries).
+        _documents_updated: >-
+          {%- set output = [] -%}
+          {%- for document in _documents -%}
+            {%- if document.kind is defined and document.kind == 'ClusterConfiguration' -%}
+              {{- output.append(document | combine(_update, recursive=true)) -}}
+            {%- else -%}
+              {{- output.append(document) -}}
+            {%- endif -%}
+          {%- endfor -%}
+          {{- output -}}
+
+# The `kubeadm upgrade` command can be executed with or without a config file.
+# If the kubeadm-config.yml file does not exists, then we at least patch the kubeadm-config configmap.
+- when:
+    - stat_etcd_encryption_config.stat.exists
+    - not stat_kubeadm_config.stat.exists
+  run_once: true  # makes no sense to execute it more than once (would be redundant)
+  block:
+    - name: Load the kubeadm-config configmap
+      shell: |
+        kubectl get configmap kubeadm-config \
+          --namespace kube-system \
+          --output yaml
+      args:
+        executable: /bin/bash
+      environment:
+        KUBECONFIG: &KUBECONFIG /etc/kubernetes/admin.conf
+      register: shell_kubeadm_configmap
+      changed_when: false
```

Comment:
Please add logic to check whether state was changed or skip this task based on output of `kubeadm config view`

Reply:
@to-bar please check this commit https://github.com/epiphany-platform/epiphany/pull/1427/commits/29bb389f9bd431f3ee239e435bd781708716fdfa I hope it solves the issue :+1:.